src/lib/math/pcurves/pcurves_algos.h

```diff
@@ -0,0 +1,282 @@
+/*
+* (C) 2024,2025 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_PCURVES_ALGOS_H_
+#define BOTAN_PCURVES_ALGOS_H_
+
+#include <botan/types.h>
+#include <botan/internal/ct_utils.h>
+#include <span>
+#include <vector>
+
+namespace Botan {
+
+/**
+* Field inversion concept
+*
+* This concept checks if the FieldElement supports fe_invert2
+*/
+template <typename C>
+concept curve_supports_fe_invert2 = requires(const typename C::FieldElement& fe) {
+   { C::fe_invert2(fe) } -> std::same_as<typename C::FieldElement>;
+};
+
+/**
+* Field inversion
+*
+* Uses the specialized fe_invert2 if available, or otherwise the standard
+* (FLT-based) field inversion.
+*/
+template <typename C>
+inline constexpr auto invert_field_element(const typename C::FieldElement& fe) {
```

Comment:
```suggestion
template <typename C>
constexpr auto invert_field_element(const typename C::FieldElement& fe) {
```

... templates don't need inlines. Also below.

Reply:
Templates are implicitly inline for linkage purposes, but do compilers use `inline` to guide whether code is actually *inlined* or not? I would think they do. (Just checked; they do)